src/controllers/expensesController.js

```diff
@@ -0,0 +1,107 @@
+const expensesService = require('../services/expensesService');
+const userService = require('../services/userService');
+
+const get = (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  if (!userId && !categories && (!from || !to)) {
+    res.send(expensesService.getAllExpenses());
+  }
+
+  const normalizedCategories = expensesService.normalizeCategory(categories);
+
+  let filteredExpenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(userId);
+  }
+
+  if (normalizedCategories) {
+    filteredExpenses =
+      expensesService.filterExpensesByCategory(normalizedCategories);
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(fromDate, toDate);
+  }
```

Comment:
There is a potential performance issue here. You are calling `getAllExpenses` method at the beginning and then possibly filtering the result multiple times. This might be inefficient if there are a lot of expenses. It would be better to build a single filter condition and apply it once. Also, it's better to retrieve data from the database only when you have all filter conditions.